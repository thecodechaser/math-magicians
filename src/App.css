 @import url("https://fonts.googleapis.com/css2?family=Poppins&display=swap");

/* universal-style */

body {
  font-family: "Poppins", sans-serif;
  background-color: #a8dadc;
}
 
 /* style for header/js */

 .header {
     display: flex;
     justify-content: space-around;
 }

 .nav-menu {
     display: flex;
     list-style-type: none;
     gap: 20px;
 }

 .nav-item {
     font-size: 18px;
     text-decoration: none;
 }

  /* style for home.js */

  .home-main {
      display: flex;
      flex-direction: column;
      align-items: center;
  }

  .home-heading {
      font-size: 25px;
  }

  .home-para {
      font-size: 18px;
      margin: 50px 200px;
  }

/* style for calculator.js */

.cal-main {
    display: flex;
    justify-content: center;
    margin-top: 100px;
  }

  .cal-heading {
      margin-right: 50px;
      font-size: 20px;
      font-weight: 600;
  }
  
  .output {
    display: flex;
    justify-content: flex-end;
    width: 400px;
    background-color: gray;
  }
  
  .output-p {
    margin-right: 5px;
    font-size: 20px;
    color: #fff;
  }
  
  .cal-btns {
    display: flex;
  }
  
  .btn {
    background-color: #fff;
    padding: 20px;
    width: 100px;
    border: 1px solid rgb(219, 212, 212);
    cursor: pointer;
    font-weight: bold;
  }
  
  .btn.last {
    background-color: rgb(255, 166, 0);
  }
  
  .btn.zero {
    width: 200px;
  }

  /* style for quote.js */

 .quote-main {
     display: flex;
     justify-content: center;
 }

 .quote {
     font-size: 20px;
     font-weight: 600;
     margin: 150px 200px;
 }
  
