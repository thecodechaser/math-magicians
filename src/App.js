import react from 'react';
import './App.css';
import Calculator from './components/Calculator';

class App extends react.Component {
  render() {
    return <Calculator />;
  }
}

export default App;
