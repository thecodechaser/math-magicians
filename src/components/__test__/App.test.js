import { render, fireEvent, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../../App';
import '@testing-library/jest-dom';

describe('App tests', () => {
  test('should render', () => {
    const app = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    expect(app).toMatchSnapshot();
  });
  test('should navigate to calculator page and find the title', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    const calcultorPage = screen.getByRole('link', { name: /calculator/i });
    fireEvent.click(calcultorPage);

    expect(
      screen.getByRole('heading', { name: /let's do some math!/i }),
    ).toBeInTheDocument();
  });

  test('should navigate to Quote page and find the find the quote', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    const QuotePage = screen.getByRole('link', { name: /quote/i });
    fireEvent.click(QuotePage);

    const quote = screen.getByText(
      /without mathematics, thereâ€™s nothing you can do\. everything around you is mathematics\. everything around you is numbers\./i,
    );

    expect(quote).toBeInTheDocument();
  });
});
