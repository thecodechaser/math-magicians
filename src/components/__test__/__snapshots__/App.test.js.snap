// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <h1
          class="header-heading"
        >
          Math Magicians
        </h1>
        <nav>
          <ul
            class="nav-menu"
          >
            <li>
              <a
                aria-current="page"
                class="nav-item active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="nav-item"
                href="/calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                class="nav-item"
                href="/quote"
              >
                Quote
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <main
        class="home-main"
      >
        <h1
          class="home-heading"
        >
          Welcome to our page!
        </h1>
        <p
          class="home-para"
        >
          End of the day, life itself is a math problem. Everything can be explained through math, including the rotation of our Earth or creation of the human kind. You will also find some inspirational math quotes for students with donuts
        </p>
        <p
          class="home-para"
        >
          There are certain calculations I should like to make with you, To be sure that your deductions will be logical and true; And remember, 'Patience, Patience,' is the watchword of a sage, Not to-day nor yet to-morrow can complete a perfect age.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <h1
        class="header-heading"
      >
        Math Magicians
      </h1>
      <nav>
        <ul
          class="nav-menu"
        >
          <li>
            <a
              aria-current="page"
              class="nav-item active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="nav-item"
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              class="nav-item"
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="home-main"
    >
      <h1
        class="home-heading"
      >
        Welcome to our page!
      </h1>
      <p
        class="home-para"
      >
        End of the day, life itself is a math problem. Everything can be explained through math, including the rotation of our Earth or creation of the human kind. You will also find some inspirational math quotes for students with donuts
      </p>
      <p
        class="home-para"
      >
        There are certain calculations I should like to make with you, To be sure that your deductions will be logical and true; And remember, 'Patience, Patience,' is the watchword of a sage, Not to-day nor yet to-morrow can complete a perfect age.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
